packages/slider/src/HandleController.ts

```diff
@@ -0,0 +1,371 @@
+        }
+        const rect = this.boundingClientRect;
+        const minOffset = rect.left;
+        const offset = event.clientX;
+        const size = rect.width;
+
+        const percent = (offset - minOffset) / size;
+        const value = this.host.min + (this.host.max - this.host.min) * percent;
+
+        return this.host.isLTR ? value : this.host.max - value;
+    }
+
+    private normalizeValue(value: number): number {
+        return (value - this.host.min) / (this.host.max - this.host.min);
+    }
+
+    public renderHandle(model: SliderHandle): TemplateResult {
+        const isActive = this.activeHandle === model.handleName;
+        const classes = ['handle'];
+        if (this.draggingHandle === model) {
+            classes.push('dragging');
+        }
+        if (model.highlight) {
+            classes.push('handle-highlight');
+        }
+        return html`
+            <div
+                class=${classes.join(' ')}
+                name=${model.handleName}
+                style=${this.handleStyle(model)}
+                @manage=${streamingListener(
+                    {
+                        type: 'pointerdown',
+                        fn: (e) => this.handlePointerdown(e, model),
+                    },
+                    {
+                        type: 'pointermove',
+                        fn: (e) => this.handlePointermove(e, model),
+                    },
+                    {
+                        type: ['pointerup', 'pointercancel'],
+                        fn: (event) => this.handlePointerup(event, model),
+                    }
+                )}
+                role="presentation"
+            >
+                <input
+                    type="range"
+                    class="input"
+                    id=${ifDefined(isActive ? 'input' : undefined)}
+                    value=${model.value}
+                    step=${this.host.step}
+                    min=${this.host.min}
+                    max=${this.host.max}
+                    aria-disabled=${ifDefined(
+                        this.host.disabled ? 'true' : undefined
+                    )}
+                    aria-valuetext=${'' /*this.ariaValueText*/}
+                    @change=${(event: Event) =>
+                        this.onInputChange(event, model)}
+                    @focus=${(event: Event) => this.onInputFocus(event, model)}
+                    @blur=${() => this.onInputBlur(model)}
+                />
+            </div>
+        `;
+    }
+
+    public render(): TemplateResult[] {
+        return Array.from(this.model.values()).map((model) =>
+            this.renderHandle(model)
+        );
+    }
+
+    public trackSegments(): [number, number][] {
+        const values = [...this.model.values()].map((item) => item.value);
+        values.sort((a, b) => a - b);
+        const normalizedValues = values.map((value) =>
+            this.normalizeValue(value)
+        );
+        normalizedValues.unshift(0);
```

Comment:
Lets say we have a slider with min=0 and max=10 and handles at 3 and 6. There will be three track segments: `[0,3], [3,6], and [6,10]`. Normalized (between 0 and 1) this will look like: `[0,0.3], [0.3,0.6], and [0.6,1]`.

The `normalizedValues.unshift(0)` puts the starting value of `0` into the list of segment start values. Then, we iterate over the start values of the segment and product tuples of the `[start, end]` values.

I will change a variable name and add a comment to make it more clear